otcextensions/osclient/mrs/v1/cluster.py

```diff
@@ -0,0 +1,417 @@
+            help=_('Cluster version.')
+        )
+        parser.add_argument(
+            '--flavor',
+            metavar='<flavor>',
+            help=_('Flavor ID.')
+        )
+        parser.add_argument(
+            '--status',
+            metavar='{' + ','.join(CLUSTER_STATES) + '}',
+            type=lambda s: s.lower(),
+            choices=CLUSTER_STATES,
+            help=_('Cluster status filter.')
+        )
+        parser.add_argument(
+            '--availability_zone',
+            metavar='<availability_zone>',
+            help=_('Availability zone.')
+        )
+        parser.add_argument(
+            '--limit',
+            metavar='<limit>',
+            type=int,
+            help=_('Number of entries to display.')
+        )
+        parser.add_argument(
+            '--marker',
+            metavar='<marker>',
+            help=_('ID of the last record on the previous page.')
+        )
+        parser.add_argument(
+            '--changes_since',
+            metavar='<changes_since>',
+            help=_('Filters the response by a date and time stamp when the '
+                   'MRS last changed status. Format: '
+                   'CCYY-MM-DDThh:mm:ss+hh:mm')
+        )
+        return parser
+
+    def take_action(self, parsed_args):
+        client = self.app.client_manager.mrs
+
+        query = {}
+
+        if parsed_args.id:
+            query['id'] = parsed_args.id
+        if parsed_args.name:
+            query['name'] = parsed_args.name
+        if parsed_args.flavor:
+            query['flavor'] = parsed_args.flavor
+        if parsed_args.status:
+            query['status'] = parsed_args.status
+        if parsed_args.cluster_type:
+            query['cluster_type'] = parsed_args.cluster_type
+        if parsed_args.version:
+            query['version'] = parsed_args.version
+        if parsed_args.availability_zone:
+            query['availability_zone'] = parsed_args.availability_zone
+        if parsed_args.limit:
+            query['limit'] = parsed_args.limit
+        if parsed_args.marker:
+            query['marker'] = parsed_args.marker
+        if parsed_args.changes_since:
+            query['changes_since'] = parsed_args.changes_since
+
+        data = client.clusters(**query)
+
+        table = (self.columns,
+                 (utils.get_item_properties(
+                     s, self.columns, formatters=_formatters
+                 ) for s in data))
+        return table
+
+
+class ListClusterHost(command.Lister):
+    _description = _('Show the MRS Cluster Hosts')
+    columns = (
+        'id', 'name', 'status', 'flavor', 'type',
+        'ip', 'mem', 'cpu', 'data_volume_size'
+    )
+
+    def get_parser(self, prog_name):
+        parser = super(ListClusterHost, self).get_parser(prog_name)
+
+        parser.add_argument(
+            'cluster_id',
+            metavar='<cluster_id>',
+            help=_('id of the cluster.')
+        )
+
+        # parser.add_argument(
```

Comment:
There are quite a lot of possible list filters. Let's add at least the most reasonables